service/resource/endpoint/current.go

```diff
@@ -19,8 +19,8 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 
 	r.logger.Log("pod", pod.GetName(), "debug", "looking for annotations on pod")
 	serviceNamespace := pod.GetNamespace()
-	_, serviceName, err := getAnnotations(*pod, IPAnnotation, ServiceAnnotation)
-	if IsMissingAnnotationError(err) {
+	podIP, serviceName, err := getAnnotations(*pod, IPAnnotation, ServiceAnnotation)
+	if IsMissingAnnotationError(err) || podIP == "" {
```

Comment:
1. I get what you want to do. I do not understand why this can happen though. Do we know that? 
2. I think a better approach to realize this here would be to check the value provided by the found annotation within `getAnnotations`. This function should then return an error the caller can check for.

Reply:
1. got sorted after I read the PR in kvm-operator. :D

Reply:
2. is still valid for me personally because I think it leads to better software design.

Reply:
this make sense